main.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017 The Alvalor Authors
+//
+// This file is part of Alvalor.
+//
+// Alvalor is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Alvalor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with Alvalor.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"os"
+	"os/signal"
+	"sync"
+
+	"github.com/alvalor/alvalor-go/network"
+	"github.com/rs/zerolog"
+)
+
+func main() {
+
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt)
+
+	log, err := zerolog.New()
+	if err != nil {
+		os.Exit(1)
+	}
+
+	wg := &sync.WaitGroup{}
+	network.New(log, wg)
```

Comment:
I think that calls to go handleDropping/handleDialing/handleServing could be extracted to some network.Start() function. This would make interface more explicit. I personally don't recommend starting actual work in the constructors.

Reply:
I don't necessarily agree. Does a network manager who is not running anything make sense to have? It would be in an invalid state, pretty much. So I would argue there is no need.

Reply:
It also depends on whether it is normal practice in go. If you download other packages and they have more or less same behaviour (starting routines from constructor), then there is nothing to discuss :)